lego/apps/surveys/views.py

```diff
@@ -0,0 +1,49 @@
+from rest_framework import decorators, exceptions, status, viewsets
+from rest_framework.response import Response
+
+from lego.apps.surveys.models import Submission, Survey
+from lego.apps.surveys.serializers import (SubmissionCreateAndUpdateSerializer,
+                                           SubmissionReadDetailedSerializer,
+                                           SubmissionReadSerializer,
+                                           SurveyCreateAndUpdateSerializer,
+                                           SurveyReadDetailedSerializer, SurveyReadSerializer)
+
+
+class SurveyViewSet(viewsets.ModelViewSet):
+    queryset = Survey.objects.all().prefetch_related('questions', 'submissions')
+
+    def get_serializer_class(self):
+        if self.action in ['create', 'update', 'partial_update']:
+            return SurveyCreateAndUpdateSerializer
+        elif self.action in ['retrieve']:
+            return SurveyReadDetailedSerializer
+        return SurveyReadSerializer
+
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        survey = Survey.objects.get(instance.id)
+        survey.delete()
```

Comment:
Use `is_answered_by_someone` here? Not sure if we want that or not :stuck_out_tongue: Would definitively be nice on updates to avoid changing a survey after it's active

Reply:
I'm thinking it should be possible for someone who's able to delete surveys to do so whenever they want. There should be some sort of validation for sure, but that can be in the front-end, the backend shouldn't really prevent it imo.